broker/pkg/storage/inmemory/inmemory.go

```diff
@@ -79,6 +80,9 @@ func (m *Memory) CreateTopic(ctx context.Context, timelineID string) error {
 	if bucketCount < 1 {
 		return ErrBucketCountLessThanOne
 	}
+
+	m.mu.Lock()
```

Comment:
The mutex should protect all usages of topics. Here you guarded against multiple CreateTopics in parallel, but we can have multiple Inserts/Deletes/Leases also.